opentelemetry-sdk/src/trace/provider.rs

```diff
@@ -36,22 +36,29 @@ static NOOP_TRACER_PROVIDER: Lazy<TracerProvider> = Lazy::new(|| TracerProvider
             span_limits: SpanLimits::default(),
             resource: Cow::Owned(Resource::empty()),
         },
+        is_shutdown: AtomicBool::new(true),
     }),
-    is_shutdown: Arc::new(AtomicBool::new(true)),
 });
 
 /// TracerProvider inner type
 #[derive(Debug)]
 pub(crate) struct TracerProviderInner {
     processors: Vec<Box<dyn SpanProcessor>>,
     config: crate::trace::Config,
+    is_shutdown: AtomicBool,
 }
 
 impl Drop for TracerProviderInner {
     fn drop(&mut self) {
-        for processor in &mut self.processors {
-            if let Err(err) = processor.shutdown() {
-                global::handle_error(err);
+        if self
+            .is_shutdown
+            .compare_exchange(false, true, Ordering::SeqCst, Ordering::SeqCst)
+            .is_ok()
+        {
```

Comment:
the logic should be
drop() 
{
1. check if already shutdown. if yes, do nothing and return
2. if no, set the is_shutdown to false, and then call shutdown
}

This PR is not doing 1 I think...

Reply:
Cancel that. This should be same as https://github.com/open-telemetry/opentelemetry-rust/pull/2195/files#r1797436168
i.e Drop should just call shutdown() if not already shutdown, and let shutdown() handle the details on how to do shutdown etc.

Reply:
> check if already shutdown. if yes, do nothing and return

compare_exchage does check if already shutdown.

 > This should be same as https://github.com/open-telemetry/opentelemetry-rust/pull/2195/files#r1797436168

same comment - https://github.com/open-telemetry/opentelemetry-rust/pull/2195/files#r1797443148 :)

Reply:
Added private TracerProviderInner::Shutdown() to be called from LoggerProvider::Shutdown and  LoggerProviderInner::Drop